lifelines/utils/sklearnAdapter.py

```diff
@@ -0,0 +1,84 @@
+import inspect
+import pandas
+from sklearn.base import BaseEstimator, RegressorMixin, MetaEstimatorMixin, MultiOutputMixin
+from . import concordance_index
+
+
+def filterKwArgs(f, kwargs):
+    s = inspect.getfullargspec(f)
+    if s.varkw:
+        return kwargs
+    else:
+        allArgs = set()
+        if s.args:
+            allArgs |= set(s.args)
+        if s.kwonlyargs:
+            allArgs |= set(s.kwonlyargs)
+        fedArgs = set(kwargs)
+        redundantArgs = fedArgs - allArgs
+        presentArgs = fedArgs & allArgs
+        #if redundantArgs:
+        #    warnings.warn("Following args are redundant for "+str(f)+str(inspect.signature(f))+": "+repr(redundantArgs))
+        res = {k: kwargs[k] for k in presentArgs}
+        return res
+
+
+class LifelinesSKLearnAdapter(BaseEstimator, MetaEstimatorMixin, RegressorMixin):
+    __slots__ = ("fitter", "params", "yArgName", "eventArgName", "remap", "concordanceRemap")
+    def __init__(self, fitter, params, yArgName="duration_col", eventArgName="event_col", remap=None, concordanceRemap=None):
```

Comment:
can you add a doc string for this class? 

Also, can we use snakecase for the parameter names?

Reply:
The same. Feel free to rename them the way you like. It is your lib, not mine.